jpa/hibernate5_3/src/main/java/org/jboss/as/jpa/hibernate5/service/WildFlyCustomJtaPlatform.java

```diff
@@ -22,39 +22,69 @@
 
 import org.hibernate.engine.transaction.jta.platform.internal.JBossAppServerJtaPlatform;
 import org.hibernate.engine.transaction.jta.platform.internal.JtaSynchronizationStrategy;
+import org.jboss.as.jpa.hibernate5.JpaLogger;
 
 /**
- * WildFlyCustomJtaPlatform
+ * WildFlyCustomJtaPlatform can obtain the JTA TransactionSynchronizationRegistry to be used by
+ * Hibernate ORM JPA + native applications.
+ * For JPA applications, we could of passed the TransactionSynchronizationRegistry into the
+ * constructor but Hibernate native apps wouldn't be able to do that, so this covers all app types.
  *
  * @author Scott Marlow
  */
-public class WildFlyCustomJtaPlatform extends JBossAppServerJtaPlatform {
+public class WildFlyCustomJtaPlatform extends JBossAppServerJtaPlatform implements JtaSynchronizationStrategy {
 
-    private static final String TSR_NAME = "java:jboss/TransactionSynchronizationRegistry";
+    // The 'transactionSynchronizationRegistry' used by JPA container managed applications,
+    // is reset every time the Transaction Manager service is restarted,
+    // as (application deployment) JPA persistence unit service depends on the TM service.
+    // For this reason, the static 'transactionSynchronizationRegistry' can be updated.
+    // Note that Hibernate native applications currently have to be (manually) restarted when the TM
+    // service restarts, as native applications do not have WildFly service dependencies set for them.
+    private static volatile TransactionSynchronizationRegistry transactionSynchronizationRegistry;
 
+    // JtaSynchronizationStrategy
     @Override
-    protected JtaSynchronizationStrategy getSynchronizationStrategy() {
-        final TransactionSynchronizationRegistry transactionSynchronizationRegistry =
-                locateTransactionSynchronizationRegistry();
-
-        return new JtaSynchronizationStrategy() {
+    public void registerSynchronization(Synchronization synchronization) {
+        if (transactionSynchronizationRegistry == null) {
+            throw JpaLogger.JPA_LOGGER.nullTransactionSynchronizationRegistry();
+        }
+        transactionSynchronizationRegistry.registerInterposedSynchronization(synchronization);
```

Comment:
Not a big deal at all, but I think at some point @dmlloyd was wanting all null checks to use `org.wildfly.common.Assert.checkNotNullParam()`. Sin this case it could just be:
```
transactionSynchronizationRegistry.registerInterposedSynchronization(Assert.checkNotNullParam("synchronization", synchronization);
```

Reply:
I didn't know we have a org.wildfly.common.Assert class :), great idea!  I'll switch to that, much better!

Reply:
Switched to our Assert class.